flang/lib/Lower/ConvertExpr.cpp

```diff
@@ -2008,23 +2065,173 @@ class ArrayExprLowering {
     return x.Rank() != 0;
   }
 
-  // Generate an error message, but no code.
-  CC errorCC(llvm::StringRef msg) {
-    mlir::emitError(getLoc(), msg);
-    return [](IterSpace) { return mlir::Value{}; };
+  // Attribute for an alloca that is a trivial adaptor for converting a value to
+  // pass-by-ref semantics for a VALUE parameter. The optimizer may be able to
+  // eliminate these.
+  mlir::NamedAttribute getAdaptToByRefAttr() {
+    return {mlir::Identifier::get("adapt.valuebyref", builder.getContext()),
+            builder.getUnitAttr()};
+  }
```

Comment:
It seems like one could write a performance pass that promotes VALUE attribute semantics (pass by reference to copy) to true pass by value in some cases. This is just used to flag the opportunity, and I figured was otherwise harmless.